### PR TITLE
[AIRFLOW-5571] Fix bug that kubernetes getting log error will stop pod unexpectedly

### PR DESCRIPTION
**Pull request by icyfox-bupt**
…d run.

- _monitor_pod function didn't catch the error for getting log, then pod
will stop unexpectedly, add a try-catch here.

Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-5571/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5571
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

In my project, I manage 200+ jobs in kubernetes, as I work on, I found a critical bug for kubernetes operator.

In pod_launcher.py, _monitor_pod function:
```
        if get_logs:
            logs = self.read_pod_logs(pod) # Here has a retry.
            for line in logs:  # But exception throw from here!
                self.
```
in above code, logs is a HttpResponse, as it implemented _iter_() function, you can use for loop to print the lines. In the other words, here use a http long connection to get endless log.

There is only try catch over self.read_pod_logs, however, If the network is disconnected or jitter occurs, for loop will throw error.

As I have 200+ job run everyday, I can get 4~5 errors everyday, and each error will let monitor think the pod is down, and then mark the task as failed, then retry it. This eventually lead to data error.

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: Add a try-catch here and test is not easy to add.

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/kubernetes/pod_launcher.py [hunk 1]
```diff
@@ -123,8 +124,13 @@ def run_pod(
     def _monitor_pod(self, pod: V1Pod, get_logs: bool) -> Tuple[State, Optional[str]]:
         if get_logs:
             logs = self.read_pod_logs(pod)
-            for line in logs:
-                self.log.info(line)
+            # Catch exception as for loop on HttpResponse may corrupt.
+            try:
+                for line in logs:
+                    self.log.info(line)
+            except ProtocolError as e:
```
Comment (ashb) [inline airflow/kubernetes/pod_launcher.py:131]: I guess this is better than nothing (and better than failing) but would it be possible to provide the `timestamps=True` (which we filter back out again) and use `since_time` to start-off again where we left off.

**Comment by OmerJog** [comment]
@icyfox-bupt are you still working on this PR?

**Comment by icyfox-bupt** [comment]
> @icyfox-bupt are you still working on this PR?

sorry for late, I would like to finish this soon.

**Comment by icyfox-bupt** [comment]
After some careful analysis, I decided to close this PR. The network problem should be solved from the network side. For airflow, the current error handling mechanism is sufficient. Sorry for late.

**Pull request closed without merging**